packages/core/src/analytics/dispatch.ts

```diff
@@ -0,0 +1,51 @@
+import { CoreContext } from '../context'
+import { CoreSegmentEvent, Callback } from '../events/interfaces'
+import { EventQueue } from '../queue/event-queue'
+import { isOffline } from '../connection'
+import { Emitter } from '../emitter'
+import { invokeCallback } from '../callback'
+
+type DispatchOptions = {
+  timeout?: number
+  debug?: boolean
+  callback?: Callback
+  retryQueue?: boolean
+}
+
+export async function dispatch(
+  event: CoreSegmentEvent,
+  queue: EventQueue,
+  emitter: Emitter,
+  options?: DispatchOptions
+): Promise<CoreContext> {
+  const ctx = new CoreContext(event)
+  emitter.emit('message_dispatch_pending', ctx) // TODO: inspectorHost.triggered?.(ctx as any)
```

Comment:
/cc @zikaari since this would end up impacting inspector work (though this PR doesn't change anything for AnalyticsBrowser yet)